Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -55,8 +55,6 @@ protocol PurchasesOrchestratorTests {
 
     func testSyncPurchasesPostsReceipt() async throws
 
-    func testSyncPurchasesDoesntPostReceiptAndReturnsCustomerInfoIfNoTransaction() async throws
```

Comment:
There is a method with this name in the `PurchasesOrchestratorSK1Tests.swift` file, which made me realize, we have an optimization in SK1 where we are not sending the receipt if:

- There are no transactions
- We have a cached `CustomerInfo`
- the cached CustomerInfo has an `originalPurchaseDate`

That logic is here: https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Purchasing/Purchases/PurchasesOrchestrator.swift#L1061

And we even have an SK1-specific test for that case: `testSyncPurchasesSK1PostsReceiptIfNoTransactionsAndEmptyOriginalPurchaseDate`

Would it make sense to add the same optimization to the SK2 path?

In that case, we could add back the `testSyncPurchasesDoesntPostReceiptAndReturnsCustomerInfoIfNoTransaction` test in SK2, and add a  common `testSyncPurchasesPostsReceiptIfNoTransactionsAndEmptyOriginalPurchaseDate` to the tests protocol.

Reply:
Oh, yes, that's a great optimization! Since originalPurchaseDate/originalApplicationVersion should never change, there should be no reason to push up the `AppTransaction`. I'll add in this optimization to the SK2 path! :D